### PR TITLE
GitLab Critical Patch Release: Upgrade GitLab CE to 17.1.1

### PR DESCRIPTION
**Pull request by Patrick-Remy**
https://about.gitlab.com/releases/2024/06/26/patch-release-gitlab-17-1-1-released

### CODE REVIEW

**Review by rageAgainstTheMachine99 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -2457,8 +2457,8 @@ Enable/disable rack middleware for blocking & throttling abusive requests Defaul
 
 ##### `RACK_ATTACK_WHITELIST`
 
-Always allow requests from whitelisted host.   
-This should be a valid yaml sequence of host address. Each host address string must be a valid IP address that can be passed to `IPAddr.new` of ruby. See [ruby-lang reference](https://docs.ruby-lang.org/en/3.0/IPAddr.html#method-c-new) for detail.  
+Always allow requests from whitelisted host.
+This should be a valid yaml sequence of host address. Each host address string must be a valid IP address that can be passed to `IPAddr.new` of ruby. See [ruby-lang reference](https://docs.ruby-lang.org/en/3.0/IPAddr.html#method-c-new) for detail.
```
Comment (kkimurak) [inline README.md:2461]: here: Some whitespaces at the end of the line are unintentionally removed

File: README.md [hunk 2]
```diff
@@ -808,10 +808,10 @@ These options should contain something like:
...
+GitLab adopted feature flags strategies to deploy features in an early stage of development so that they can be incrementally rolled out. GitLab administrators with access to the [Rails console](https://docs.gitlab.com/ee/administration/feature_flags.html#how-to-enable-and-disable-features-behind-flags) or the [Feature flags API](https://docs.gitlab.com/ee/api/features.html) can control them (note that `sameersbn/gitlab` is a container image that provides GitLab installations from the source).
+You can see all feature flags in GitLab at corresponding version of documentation: <https://docs.gitlab.com/ee/user/feature_flags.html>
 
-For `sameersbn/gitlab`, you can control them via environment parameter [`GITLAB_FEATURE_FLAGS_DISABLE_TARGETS`](#gitlab_feature_flags_disable_targets) and [`GITLAB_FEATURE_FLAGS_ENABLE_TARGETS`](#gitlab_feature_flags_enable_targets) in addition to the above methods.  
+For `sameersbn/gitlab`, you can control them via environment parameter [`GITLAB_FEATURE_FLAGS_DISABLE_TARGETS`](#gitlab_feature_flags_disable_targets) and [`GITLAB_FEATURE_FLAGS_ENABLE_TARGETS`](#gitlab_feature_flags_enable_targets) in addition to the above methods.
```
Comment (kkimurak) [inline README.md:814]: here: Some whitespaces at the end of the line are unintentionally removed

File: README.md [hunk 3]
```diff
@@ -825,7 +825,7 @@ services:
     - GITLAB_FEATURE_FLAGS_ENABLE_TARGETS=git_push_create_all_pipelines,build_service_proxy
 ````
 
-Once the container up, you can see following messages in container log like below.  
+Once the container up, you can see following messages in container log like below.
```
Comment (kkimurak) [inline README.md:828]: here: Some whitespaces at the end of the line are unintentionally removed

File: README.md [hunk 4]
```diff
@@ -1674,9 +1674,9 @@ The value of the `report-uri` directive in the `Content-Security-Policy` header
 
 ##### `GITLAB_FEATURE_FLAGS_DISABLE_TARGETS`
 
-Comma separated list of feature flag names to be disabled. No whitespace is allowed.  
-You can see all feature flags in GitLab at corresponding version of documentation: <https://docs.gitlab.com/ee/user/feature_flags.html>  
-Feature flags name and its statement will be appear to container log. Note that some of the feature flags are implicitly enabled or disabled by GitLab itself, and are not appear to container log.  
+Comma separated list of feature flag names to be disabled. No whitespace is allowed.
+You can see all feature flags in GitLab at corresponding version of documentation: <https://docs.gitlab.com/ee/user/feature_flags.html>
+Feature flags name and its statement will be appear to container log. Note that some of the feature flags are implicitly enabled or disabled by GitLab itself, and are not appear to container log.
```
Comment (kkimurak) [inline README.md:1679]: here: Some whitespaces at the end of the line are unintentionally removed

**Review by kkimurak — COMMENTED** [review]
In README.md, Some whitespaces at the end of the line are unintentionally removed. Perhaps your editor did it automatically.  
I would appreciate it if you would commit only the intended changes (that is, changes that are consistent with the purpose of the pull request).  

There are no coding guidelines for this repository, and (as far as I know) no one checked them in previous reviews. And (luckyly) these changes are not so critical.  

So, feel free to ignore them. I made this a comment and did not set it to "request for change".

**Comment by sachilles** [comment]
@kkimurak You are right. However, we have to have some style / code guidelines in order to fullfill them by our own as well.

**Comment by Patrick-Remy** [comment]
Thank you for merging.

To be honest, the only solution for all of this is to automate the whole process, at least by adding a shell script for half-automation. For most updates, it seems to me just to be:
- checking for latest versions of the deps
- maybe check OS base image version
- search and replace 
- add a few lines changelog

And one day add a bit that opens a MR if there is a new GitLab version.

And for "code style", maybe an editorconfig file shortly helps. Most of the IDEs will respect them.

**Comment by donbowman** [comment]
I'm not clear, this was merged and a tag created, but no build? was that normally automatic?

**Pull request merged**